### PR TITLE
quicklogic: use external arch and rr_graph

### PR DESCRIPTION
**Pull request by acomodi**
This PR allows to fetch an external RR graph and arch.xml, which comes from a submodule.
It also removes the architecture, which is now taken from a third_party directory and updates conda packages

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -8,11 +8,12 @@ dependencies:
...
   - litex-hub::iverilog=s20150603_0957_gad862020=20201120_145821
   - litex-hub::openocd=0.10.0_1514_ga8edbd020=20201119_154304
-  - litex-hub::symbiflow-yosys-plugins=1.0.0_7_263_gac47a74=20210312_125539
+  - litex-hub::symbiflow-yosys-plugins=1.0.0_7_274_g4b28551=20210318_102115
   - litex-hub::prjxray-tools=0.1_2842_g6867429c=20210301_104249
   - litex-hub::prjxray-db=0.0_248_g2e51ad3=20210312_125539
-  - litex-hub::vtr-optimized=8.0.0_3445_ga806b1609=20210312_125539
+  - litex-hub::vtr-optimized=8.0.0_3452_ge7d45e013=20210318_102115
   - litex-hub::yosys=0.9_5266_g0fb4224e=20210301_104249_py37
+  - litex-hub::symbiflow-yosys=0.9_0_g40d9e120=20210317_102909_libffi33
```
Comment (litghost) [inline environment.yml:16]: Why are we using the `symbiflow-yosys` again?

File: quicklogic/common/cmake/quicklogic_qlf_arch.cmake [hunk 2]
```diff
@@ -38,14 +38,11 @@ function(QUICKLOGIC_DEFINE_QLF_ARCH)
...
 
-    RR_PATCH_TOOL
-      ${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/qlf_k4n8/utils/fixup_rr_graph.py
-    RR_PATCH_CMD "\${CMAKE_COMMAND} -E env \
-      PYTHONPATH=${symbiflow-arch-defs_SOURCE_DIR}/utils:$PYTHONPATH:${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/common/utils \
-      \${PYTHON3} \${RR_PATCH_TOOL} \
-          --rr-graph-in \${OUT_RRXML_VIRT} \
-          --rr-graph-out \${OUT_RRXML_REAL}"
-  
+    RR_PATCH_TOOL dummy
```
Comment (litghost) [inline quicklogic/common/cmake/quicklogic_qlf_arch.cmake:41]: This seems wrong?

File: quicklogic/common/cmake/quicklogic_qlf_device.cmake [hunk 3]
```diff
@@ -57,27 +109,25 @@ function(QUICKLOGIC_DEFINE_QLF_DEVICE)
...
-  set(DEVICE_RR_PATCH_EXTRA_ARGS "")
-
   # Define the device
   define_device(
     DEVICE ${DEVICE}
     ARCH ${ARCH}
     DEVICE_TYPE ${DEVICE_TYPE}
     PACKAGES ${PACKAGE}
     ROUTE_CHAN_WIDTH ${ROUTE_CHAN_WIDTH}
-    WIRE_EBLIF ${symbiflow-arch-defs_SOURCE_DIR}/common/wire.eblif
```
Comment (litghost) [inline quicklogic/common/cmake/quicklogic_qlf_device.cmake:78]: Without `WIRE_ELBIF`, how is the lookahead generated?

Comment (mkurc-ant) [reply quicklogic/common/cmake/quicklogic_qlf_device.cmake:78]: If you don't specify a "wire.eblif" explicitly the default one is used: https://github.com/antmicro/symbiflow-arch-defs/blob/712493b81351450f5244814d00369020ef976171/common/cmake/devices.cmake#L556-L560

**Comment by mkurc-ant** [comment]
https://github.com/SymbiFlow/symbiflow-arch-defs/pull/2119 Got merged. This one is next.

**Comment by mkurc-ant** [comment]
Kokoro CI failed again with "tool failed" status ("Architecture Definitions (Presubmit) — Build infrastructure error."). Retriggering.

**Comment by mkurc-ant** [comment]
Kokoro CI failed n-th time. Retriggering again.

**Comment by mkurc-ant** [comment]
For unknown reason each Kokoro CI reports `[ERROR] Config validation failed!`

**Comment by mkurc-ant** [comment]
https://github.com/SymbiFlow/symbiflow-arch-defs/issues/2124

**Comment by kgugala** [comment]
xc7 install CI hanged, the rest is green. Since this PR does not touch xc7 install related code I'm merging it to not stall it further

**Pull request merged**